xalia/AtSpi2/ComponentProvider.cs

```diff
@@ -77,10 +80,29 @@ public override UiDomValue EvaluateIdentifier(UiDomElement element, string ident
                     if (AbsPosKnown)
                         return new UiDomInt(AbsHeight);
                     return UiDomUndefined.Instance;
+                case "spi_grab_focus":
+                    return new UiDomRoutineAsync(element, "spi_grab_focus", GrabFocus);
             }
             return UiDomUndefined.Instance;
         }
 
+        private static async Task GrabFocus(UiDomRoutineAsync obj)
+        {
+            var provider = obj.Element.ProviderByType<ComponentProvider>();
+            try
+            {
+                var success = await CallMethod(provider.Connection.Connection, provider.Peer, provider.Path,
+                    IFACE_COMPONENT, "GrabFocus", ReadMessageBoolean);
+                if (!success)
+                    Console.WriteLine($"WARNING: spi_set_focus failed for {obj.Element}");

```

Comment:
not a functional issue, but it's `spi_grab_focus` everywhere else, so this should probably be `spi_grab_focus` and not `spi_set_focus`.

Reply:
Thanks.